packages/core/test/synchronizerAttesters.test.ts

```diff
@@ -0,0 +1,56 @@
+// @ts-ignore
+import { ethers } from 'hardhat'
+import { expect } from 'chai'
+import { EPOCH_LENGTH } from '@unirep/contracts'
+import { defaultProver } from '@unirep/circuits/provers/defaultProver'
+import { deployUnirep } from '@unirep/contracts/deploy'
+import { Synchronizer } from '../src/Synchronizer'
+
+describe('Synchronizer watch multiple attesters', function () {
+    this.timeout(0)
+
+    let unirepContract
+
+    before(async () => {
+        const accounts = await ethers.getSigners()
+        unirepContract = await deployUnirep(accounts[0])
+        for (let x = 0; x < 10; x++) {
+            const attester = accounts[x]
+            await unirepContract
+                .connect(attester)
+                .attesterSignUp(EPOCH_LENGTH)
+                .then((t) => t.wait())
+        }
```

Comment:
I prefer that a synchronizer can start first
then attesters sign up and the event should be caught by the synchronizer.

Reply:
There's no different between starting the synchronizer before emitting the event or after. In either case the codepath is the same.

Reply:
I did add a test for `AttesterSignedUp`